src/Workspaces/Remote/Core/AbstractAssetProvider.cs

```diff
@@ -19,28 +19,29 @@ internal abstract class AbstractAssetProvider
     /// <summary>
     /// return data of type T whose checksum is the given checksum
     /// </summary>
-    public abstract ValueTask<T> GetAssetAsync<T>(Checksum checksum, CancellationToken cancellationToken);
+    public abstract ValueTask<T> GetAssetAsync<T>(ProjectId? hintProject, Checksum checksum, CancellationToken cancellationToken);
 
     public async Task<SolutionInfo> CreateSolutionInfoAsync(Checksum solutionChecksum, CancellationToken cancellationToken)
     {
-        var solutionChecksums = await GetAssetAsync<SolutionStateChecksums>(solutionChecksum, cancellationToken).ConfigureAwait(false);
-        var solutionAttributes = await GetAssetAsync<SolutionInfo.SolutionAttributes>(solutionChecksums.Attributes, cancellationToken).ConfigureAwait(false);
+        var solutionChecksums = await GetAssetAsync<SolutionStateChecksums>(hintProject: null, solutionChecksum, cancellationToken).ConfigureAwait(false);
+        var solutionAttributes = await GetAssetAsync<SolutionInfo.SolutionAttributes>(hintProject: null, solutionChecksums.Attributes, cancellationToken).ConfigureAwait(false);
```

Comment:
no hints here (because this is solution info, not project info).  However, with teh reordering we did in the solution-finding code, this will all be cheap.  that's because we search the direct (non-project-child) info of the solution first, before recursing.  So we'll always find this data in O(1) time, instead of O(solution) time.